mitiq/pec/channels.py

```diff
@@ -139,53 +135,72 @@ def local_depolarizing_kraus(
     depolarizing channels acting on each qubit.
     """
     local_kraus = global_depolarizing_kraus(noise_level, num_qubits=1)
-    return _tensor_product_kraus(local_kraus, num_channels=num_qubits)
+    return [
+        tensor_product(*tup) for tup in product(local_kraus, repeat=num_qubits)
+    ]
 
 
 def amplitude_damping_kraus(
     noise_level: float, num_qubits: int
 ) -> List[np.ndarray]:
     """Returns the Kraus operators of an amplitude damping
-    channel at a given noise level. If num_qubits > 1, the Kraus operators
+    channel at a given noise level. If ``num_qubits > 1``, the Kraus operators
     corresponding to tensor product of many single-qubit amplitude damping
     channels are returned.
     """
     noisy_op = AmplitudeDampingChannel(noise_level)
     local_kraus = list(channel(noisy_op))
-    return _tensor_product_kraus(local_kraus, num_channels=num_qubits)
+    return [
+        tensor_product(*tup) for tup in product(local_kraus, repeat=num_qubits)
+    ]
 
 
-def matrix_to_vector(density_matrix: np.ndarray) -> np.array:
-    """Reshapes a density matrix into a vector, according to the rule:
-    |i><j| --> |i,j>.
+def matrix_to_vector(density_matrix: np.ndarray) -> np.ndarray:
+    r"""Reshapes a :math:`d \times d` density matrix into a
+    :math:`d^2`-dimensional state vector, according to the rule:
+    :math:`|i \rangle\langle j| \rightarrow |i,j \rangle`.
     """
     return density_matrix.flatten()
 
 
-def vector_to_matrix(vector: np.ndarray) -> np.array:
-    """Reshapes a vector to a density matrix, according to the rule:
-    |i,j> --> |i><j|.
+def _safe_sqrt(perfect_square: int) -> int:
+    """Takes the square root of the input integer and
+    raises an error if the input is not a perfect square."""
+    square_root = int(np.round(np.sqrt(perfect_square)))
+    if square_root ** 2 != perfect_square:
+        raise ValueError(
+            "The expected dimension of the input matrix (or array) must be a"
+            " square number but dimension is {perfect_square}."
+        )
```

Comment:
nit: This assumes `perfect_square` is the dimension of a matrix / array but the function is for any int. Error message could originate from calling function.

Reply:
I added a custom error string which should have fixed this problem.